ftrip.io.catalog-service/ftrip.io.catalog-service/Accommodations/UseCases/DeleteAccommodation/DeleteAccommodationRequestHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿using ftrip.io.catalog_service.Accommodations.Domain;
+using ftrip.io.framework.Contexts;
+using ftrip.io.framework.ExceptionHandling.Exceptions;
+using ftrip.io.framework.Globalization;
+using ftrip.io.framework.Persistence.Contracts;
+using MediatR;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ftrip.io.catalog_service.Accommodations.UseCases.DeleteAccommodation
+{
+    public class DeleteAccommodationRequestHandler : IRequestHandler<DeleteAccommodationRequest, Accommodation>
+    {
+        protected readonly IUnitOfWork _unitOfWork;
+        protected readonly IAccommodationRepository _accommodationRepository;
+        protected readonly IStringManager _stringManager;
+        protected readonly CurrentUserContext _currentUserContext;
+
+        public DeleteAccommodationRequestHandler(
+            IUnitOfWork unitOfWork,
+            IAccommodationRepository accommodationRepository,
+            IStringManager stringManager,
+            CurrentUserContext currentUserContext)
+        {
+            _unitOfWork = unitOfWork;
+            _accommodationRepository = accommodationRepository;
+            _stringManager = stringManager;
+            _currentUserContext = currentUserContext;
+        }
+
+        public async Task<Accommodation> Handle(DeleteAccommodationRequest request, CancellationToken cancellationToken)
+        {
+            await CheckIfCanModify(request.Id);
```

Comment:
I usually do this in the middleware/controller so I don't have to pollute business logic with this but this is also fine.